onnxruntime/core/providers/shared_library/provider_bridge_provider.cc

```diff
@@ -576,6 +577,16 @@ namespace contrib {
 Status Group::Compute(OpKernelContext* context) const { return g_host_cpu.contrib__Group__Compute(this, context); }
 Status PassThrough::Compute(OpKernelContext* context) const { return g_host_cpu.contrib__PassThrough__Compute(this, context); }
 Status YieldOp::Compute(OpKernelContext* context) const { return g_host_cpu.contrib__YieldOp__Compute(this, context); }
+
+Status AdamWOptimizerBase::PrepareForCompute(OpKernelContext* ctx, AdamWOptimizerBase::Prepare& prepare) const {
```

Comment:
I am not very familiar with this, why do we need changes to this file?

Reply:
AdamWOptimizerBase is a base class that is defined in our core lib. CUDA EP build its code in a separate lib, so there are few bridging files to let some code in CUDA EP calls into CPU EP code.